### PR TITLE
[6.14.z] Add Ruff rule group B (flake8-bugbear)

### LINKED ISSUE #13688: [Failed-AutoCherryPick] - Add Ruff rule group B (flake8-bugbear)
#### Auto-Cherry-Pick WorkFlow Failure:
- To Branch: 6.14.z
- [Failed Cherrypick Action](https://github.com/SatelliteQE/robottelo/actions/runs/7462643368)
- [Parent Pull Request](https://github.com/SatelliteQE/robottelo/pull/13491)

### PR DESCRIPTION
**Pull request by JacobCallahan**
Cherry-pick of https://github.com/SatelliteQE/robottelo/pull/13491

This change includes the rule addition and mostly manual changes to existing code in order to bring it in line with the new rules and avoid as many ignores as is reasonable.

I added a number of {{from err}} statements to the end of exceptions that currently include the error message. The error message being included in the raised string can be removed later, but keeping for consistency for now.

In cases where I raised from None, the raised exception is clear enough that the additional context of the parent exception isn't helpful.

cli/test\_ping.py has a zip that should probably be strict.

fixes #13688

### CODE REVIEW

File: tests/foreman/ui/test_contenthost.py [hunk 1]
```diff
@@ -1588,7 +1588,7 @@ def test_unset_syspurpose_attributes_cli(session, default_location, vm_module_st
         details = session.contenthost.read(vm_module_streams.hostname, widget_names='details')[
             'details'
         ]
-        for spname, spdata in DEFAULT_SYSPURPOSE_ATTRIBUTES.items():
+        for spname in DEFAULT_SYSPURPOSE_ATTRIBUTES:
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_contenthost.py:1591]:
```suggestion
        for spname in DEFAULT_SYSPURPOSE_ATTRIBUTES.values():
```

File: tests/foreman/ui/test_contenthost.py [hunk 2]
```diff
@@ -1492,7 +1492,7 @@ def test_syspurpose_attributes_empty(session, default_location, vm_module_stream
         ]
         syspurpose_status = details['system_purpose_status']
         assert syspurpose_status.lower() == 'not specified'
-        for spname, spdata in DEFAULT_SYSPURPOSE_ATTRIBUTES.items():
+        for spname in DEFAULT_SYSPURPOSE_ATTRIBUTES:
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_contenthost.py:1495]:
```suggestion
        for spname in DEFAULT_SYSPURPOSE_ATTRIBUTES.values():
```

File: robottelo/host_helpers/satellite_mixins.py [hunk 3]
```diff
@@ -235,9 +235,9 @@ def available_capsule_port(self):
         :rtype: int
         """
         port_pool_range = settings.fake_capsules.port_range
-        if type(port_pool_range) is str:
+        if isinstance(port_pool_range, list):
```
Comment (Gauravtalreja1) [inline robottelo/host_helpers/satellite_mixins.py:238]: is it correct?
```suggestion
        if isinstance(port_pool_range, str):
```

Comment (JacobCallahan) [reply robottelo/host_helpers/satellite_mixins.py:238]: nice catch!

Comment (Gauravtalreja1) [reply robottelo/host_helpers/satellite_mixins.py]: @JacobCallahan Could you fix this in master branch manually?
I'll add similar comments to your other cherrypick PRs and we can commit this suggestion in same way

**Review by Gauravtalreja1 — APPROVED** [review]

File: robottelo/host_helpers/satellite_mixins.py (hunk 3 above)
Comment (Gauravtalreja1) [reply robottelo/host_helpers/satellite_mixins.py]: manual fix required for master and 6.15.z branches*

**Pull request merged**